src/reactionsystem.jl

```diff
@@ -0,0 +1,127 @@
+using Catalyst, ModelingToolkit, Symbolics
+
+# module SBML
+# struct Model end
+# struct Reaction end
+# struct Compartment end
+# struct Species end
+# end
+
+""" ReactionSystem constructor """
+function ModelingToolkit.ReactionSystem(model::Model)  # Todo: requires unique parameters (i.e. SBML must have been imported with localParameter promotion in libSBML)
+    model = make_extensive(model)
+    model = expand_reversible(model)
+    rxs = mtk_reaction.(model.reactions)
+    t = DEFAULT_IV
+    species = [create_var(k) for k in keys(model.species)]
+    params = vcat([create_var(k) for k in keys(model.parameters)], [create_par(k) for k in keys(model.compartments)])
+    ReactionSystem(rxs,t,species,params)
+end
+
+""" ReactionSystem constructor """
+function ModelingToolkit.ReactionSystem(sbmlfile::String)
+    model = readSBML(sbmlfile)
+    ReactionSystem(model)
+end
+
+""" ODESystem constructor """
+function ModelingToolkit.ODESystem(model::Model)
+    rs = ReactionSystem(model)
+    u0map = get_u0(model)
+    parammap = get_paramap(model)
+    defaults = vcat(u0map, parammap)
+    convert(ODESystem, rs, defaults=defaults)
+end
+
+""" ODESystem constructor """
+function ModelingToolkit.ODESystem(sbmlfile::String)
```

Comment:
eventually need to fix the kwargs, so that `@named mymodel = ODESystem(sbmlfile)` will work

Reply:
where can I find documentation of the `@named` macro. I've never used it.

Reply:
```julia 
help?> @named
  Rewrite @named y = foo(x) to y = foo(x; name=:y).
```

Reply:
I think you basically just need to change this to 
`function ModelingToolkit.ODESystem(sbmlfile::String; kwargs...)`
and then pass them to the actual constructor

Reply:
Thanks!